components/core/src/clp_s/clp-s.cpp

```diff
@@ -31,6 +41,197 @@ using clp_s::cEpochTimeMax;
 using clp_s::cEpochTimeMin;
 using clp_s::CommandLineArguments;
 
+namespace {
+/**
+ * Compresses the input files specified by the command line arguments into an archive.
+ * @param command_line_arguments
+ * @return Whether compression was successful
+ */
+bool compress(CommandLineArguments const& command_line_arguments);
+
+/**
+ * Decompresses the archive specified by the given JsonConstructorOption.
+ * @param json_constructor_option
+ */
+void decompress_archive(clp_s::JsonConstructorOption const& json_constructor_option);
+
+/**
+ * Searches the given archive.
+ * @param command_line_arguments
+ * @param archive_dir
+ * @param expr The search AST
+ * @return Whether the search succeeded
+ */
+bool search_archive(
+        CommandLineArguments const& command_line_arguments,
+        std::string const& archive_dir,
+        std::shared_ptr<Expression>& expr
+);
+
+bool compress(CommandLineArguments const& command_line_arguments) {
+    clp_s::TimestampPattern::init();
```

Comment:
I think this TimestampPattern::init should be in main to be consistent with the search flow. Potentially we should just move it before any of the command-specific code -- don't think there's too much point in avoiding calling it before the decompression code.